### PR TITLE
Feature/language

### PR DESCRIPTION
**Pull request by siru-kid**
addition of Language.jsx ,LanguageForm.jsx and useLanguage.jsx(hook) and modification of AdminSettings.jsx

### CODE REVIEW

File: src/hooks/useLanguage.js [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+          setErrors((prevErrors) => {
+            return {
+              ...prevErrors,
+              ...error
+            };
+          });
+        });
+    },
+    [schema]
+  );
```
Comment (davewish) [inline src/hooks/useLanguage.js:67]: `const validateObject = useCallback((formData) => {
    return new Promise((resolve, reject) => {
      schema
        .validate(formData, { abortEarly: false })
        .then(() => {
          console.log("infor >> :Validation succeeded.");
          setErrors({});
          resolve();
        })
        .catch((error) => {
          console.log("Error >> :Validation failed.");
          const formattedErrors = error.inner.reduce((acc, err) => {
            return {
              ...acc,
              [err.path]: err.message
            };
          }, {});
          setErrors(formattedErrors);
          reject(formattedErrors);
        });
    });
  }, []);`. please update  with these

File: src/hooks/useLanguage.js [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    validateObject(langConfig);
+    axiosPrivate
+      .post(`/api/protected/${currentConfig.pageConfig}`, langConfig)
+      .then(({ data }) => {
+        console.log("saved succefylly ", data);
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+  }, [langConfig, validateObject]);
```
Comment (davewish) [inline src/hooks/useLanguage.js:126]: it is not working as expected  , please update with this

`validateObject(pageConfig)
      .then(() => {
        return axiosPrivate.post(`/api/protected/${currentConfig.pageConfig}`, data);
      })
      .then(({ data }) => {
        console.log("saved succefylly ", data);
      })
      .catch((err) => {
        console.error("error :>> ", err);
      });`

File: src/hooks/useLanguage.js [hunk 3]
```diff
@@ -0,0 +1,162 @@
...
+  }, []);
+  const handleAddModalClose = useCallback(() => {
+    setModalOpenAdd(false);
+    setLangConfig(langConfigInitial);
+  }, [setModalOpenAdd, setLangConfig]);
+
+  const saveLangeConfig = useCallback(() => {
+    validateObject(langConfig);
+    axiosPrivate
+      .post(`/api/protected/${currentConfig.pageConfig}`, langConfig)
```
Comment (davewish) [inline src/hooks/useLanguage.js:119]: please change to language API

File: src/hooks/useLanguage.js [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+      })
+      .catch((err) => {
+        console.error("error :>> ", err);
+      });
+  }, [langConfig, validateObject]);
+
+  const updateLangConfig = useCallback(() => {
+    validateObject(langConfig);
+    axiosPrivate
+      .put(`/api/protected/${currentConfig.pageConfig}`, langConfig)
```
Comment (davewish) [inline src/hooks/useLanguage.js:131]: please change with language api

File: src/hooks/useLanguage.js [hunk 5]
```diff
@@ -0,0 +1,162 @@
...
+        };
+      });
+      validateField(name, value);
+    },
+    [validateField, setLangConfig]
+  );
+
+  const populateLanguageForm = (row) => {
+    const { lang_id, lang_name, tenant_id } = row.original;
+    setLangConfig((prevState) => {
```
Comment (davewish) [inline src/hooks/useLanguage.js:85]: please let us change  this  data should come from API , there is insufficient  or inconsistent data ,
`const populatePageConfigForm = useCallback(
    (row) => {
      const { id } = row.original;
      axiosPrivate
        .get(`/api/protected/${currentConfig.pageConfig}`, {
          params: {
            id
          }
        })
        .then(({ data }) => {
          const pageConfigTemp = {
            id: data.id,
            pageType: data.page_type,
            parent: data.parent,
            headerText: data.header_text,
            language: data.language,
            imageLink: data.img_link,
            headerImage: data.header_image,
            orderNumber: data.order_number,
            description: data.description,
            name: data.name
          };
          setPageConfig((prevPageConfig) => {
            return {
              ...prevPageConfig,
              ...pageConfigTemp
            };
          });
          handleAddModalOpen("Update Page Config");
        })
        .catch((error) => {
          console.log("Error : >> ", error);
        });
    },
    [handleAddModalOpen]
  );`

**Review by davewish — CHANGES_REQUESTED** [review]

File: src/hooks/useLanguage.js (hunk 1 above)
Comment (siru-kid) [reply src/hooks/useLanguage.js]: done

File: src/hooks/useLanguage.js (hunk 2 above)
Comment (siru-kid) [reply src/hooks/useLanguage.js]: done

File: src/hooks/useLanguage.js (hunk 3 above)
Comment (siru-kid) [reply src/hooks/useLanguage.js]: done

File: src/hooks/useLanguage.js (hunk 4 above)
Comment (siru-kid) [reply src/hooks/useLanguage.js]: done

File: src/hooks/useLanguage.js (hunk 5 above)
Comment (siru-kid) [reply src/hooks/useLanguage.js:89]: okay dave but why do we need most of the data here like the "parent",headertext","language","imageLink"....etc i thought we dont need those in the language section

**Review by misghna — APPROVED** [review]

**Pull request merged**